liquibase-core/src/main/java/liquibase/lockservice/ProlongingLockService.java

```diff
@@ -0,0 +1,879 @@
+package liquibase.lockservice;
+
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import liquibase.configuration.GlobalConfiguration;
+import liquibase.configuration.LiquibaseConfiguration;
+import liquibase.database.Database;
+import liquibase.database.ObjectQuotingStrategy;
+import liquibase.database.core.DerbyDatabase;
+import liquibase.database.core.MSSQLDatabase;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.LockException;
+import liquibase.exception.UnexpectedLiquibaseException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.logging.LogFactory;
+import liquibase.snapshot.InvalidExampleException;
+import liquibase.snapshot.SnapshotGeneratorFactory;
+import liquibase.sql.Sql;
+import liquibase.sqlgenerator.SqlGeneratorFactory;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.AddColumnStatement;
+import liquibase.statement.core.CreateDatabaseChangeLogLockTableStatement;
+import liquibase.statement.core.DropTableStatement;
+import liquibase.statement.core.InitializeDatabaseChangeLogLockTableStatement;
+import liquibase.statement.core.LockDatabaseChangeLogStatement;
+import liquibase.statement.core.ProlongDatabaseChangeLogLockStatement;
+import liquibase.statement.core.RawSqlStatement;
+import liquibase.statement.core.RemoveStaleLocksStatement;
+import liquibase.statement.core.SelectFromDatabaseChangeLogLockStatement;
+import liquibase.statement.core.UnlockDatabaseChangeLogStatement;
+import liquibase.structure.core.Table;
+
+/**
+ * Prolongs the current lock every 30 seconds.
+ * <p>
+ * Locks older than 100 seconds are considered stale and will get removed.
+ * <p>
+ * Since locks get actively prolonged, do use a very high lock wait time (1 day).
+ * <p>
+ * Must <b>not</b> be used in junction with the {@link StandardLockService}!
+ * <p>
+ * Note: This is in alpha state. We do not use the time of the database server right now,
+ * but a local date, so this might not work properly when several services with very
+ * un-synchronised locks connect to the same database.
```

Comment:
It seems like it would be better to always use database time; is that not possible here?

Reply:
It's been quite a while that I wrote this, but I think it works this way: In the lock on the database, we use database time (so that all clients, no matter what local time they have, can verify if the lock is stale), but locally, we use java.util.Date to measure if the lock we just created is stale. 
Probably the comment is also obsolete now, could be I put this when I wrote the first version.
Anyways, should be analysed before integrating!